internal/safepaths/absolute.go

```diff
@@ -0,0 +1,74 @@
+package safepaths
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+// Absolute must be constructed via ParseAbsolute, or other methods in this package.
+// The zero value of Absolute will panic when String is called.
+type Absolute struct {
+	path string
+}
+
+// ParseAbsolute takes a string path that may be relative and returns
+// an Absolute that is guaranteed to be absolute, or an error.
+func ParseAbsolute(path string) (Absolute, error) {
+	path, err := filepath.Abs(path)
+	if err != nil {
+		return Absolute{}, fmt.Errorf("failed to get absolute path: %w", err)
+	}
+
+	return Absolute{path: path}, nil
+}
```

Comment:
I wonder if it is easier to understand and being more idiomatic to rename this `NewAbsolute`; it doesn't have to change but it came to mind.

Reply:
I'm open to it but I have a strong preference for `Parse` for [reasons](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/).

If you feel strongly, I'm happy for you to make the change, I agree it may be more idiomatic.